istioctl/cmd/describe.go

```diff
@@ -1186,3 +1197,94 @@ func containerReady(pod *v1.Pod, containerName string) (bool, error) {
 	}
 	return false, fmt.Errorf("no container %q in pod", containerName)
 }
+
+// describePeerAuthentication fetches all PeerAuthentication in workload and root namespace.
+// It lists the ones applied to the pod, and the current active mTLS mode.
+// When the client doesn't have access to root namespace, it will only show workload namespace Peerauthentications.
+func describePeerAuthentication(writer io.Writer, configClient istioclient.Interface, istioNamespace, workloadNamespace string, pod *v1.Pod, podsLabels k8s_labels.Set) error { // nolint: lll
+	workloadPAList, err := configClient.SecurityV1beta1().PeerAuthentications(workloadNamespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to fetch workload namespace PeerAuthentication: %v", err)
+	}
+
+	hasRootNamespaceAccess := true
+	rootPAList, err := configClient.SecurityV1beta1().PeerAuthentications(istioNamespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
```

Comment:
I think you can just return the error, it seems the only possible case that this would fail is due to some network issues (or very unlikely some RBAC issues), either case you don't need to continue the analyze because you don't have enough information to give the accurate result.

Reply:
How about is the user doesn't have root namespace access?

Reply:
then you don't need to continue the analysis since you cannot get the accurate result anyway.

Reply:
So what I did is, if user doesn't have access to root namespace, only show the matched PA in workload namespace, but will not show active mtls mode. Thoughts are that might be helpful rather than show nothing in that case.

Reply:
I think that is unnecessary and make the code complicated without much benefits.

Reply:
Sg. Still think it might be useful, but not sure how much though. Will do the error for now, we can revisit if there is need.